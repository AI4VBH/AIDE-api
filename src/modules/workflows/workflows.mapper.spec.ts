/*
 * Copyright 2022 Crown Copyright
 *
 *  Licensed under the Apache License, Version 2.0 (the "License");
 *  you may not use this file except in compliance with the License.
 *  You may obtain a copy of the License at
 *
 *  http://www.apache.org/licenses/LICENSE-2.0
 *
 *  Unless required by applicable law or agreed to in writing, software
 *  distributed under the License is distributed on an "AS IS" BASIS,
 *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *  See the License for the specific language governing permissions and
 *  limitations under the License.
 */

import {
  MonaiWorkflow,
  PagedMonaiWorkflows,
} from './monai-workflow.interfaces';
import {
  mapToPagedWorkflowsDto,
  mapToPagedWorkflowsItemDto,
} from './workflows.mapper';

describe('mapToPagedWorkflowsDto', () => {
  it.each([null, undefined])('throws exception when paged is %s', (paged) => {
    const action = () => mapToPagedWorkflowsDto(paged);

    expect(action).toThrowError('paged is null or undefined');
  });

  it('returns expected result', () => {
    const paged: PagedMonaiWorkflows = {
      pageNumber: 1,
      pageSize: 10,
      firstPage: '/page/1',
      lastPage: '/page/2',
      previousPage: null,
      nextPage: null,
      totalPages: 1,
      totalRecords: 0,
      data: [],
    };

    const result = mapToPagedWorkflowsDto(paged);

    expect(result).toMatchSnapshot();
  });
});

describe('mapToPagedWorkflowsItemDto', () => {
  it('returns expected result', () => {
    const workflow: MonaiWorkflow = {
      id: 'fb94b9da-e957-4378-a626-8f0374e60cfb',
      workflow_id: 'c7610e0b-02b7-474d-b6e5-1224debc414d',
      revision: 2,
      workflow: {
        name: 'Super Workflow',
        version: '1.0.0',
        description: 'Some description',
        informatics_gateway: {
          ae_title: '',
          data_origins: [''],
          export_destinations: [''],
        },
      },
    };

    const result = mapToPagedWorkflowsItemDto(workflow);

    expect(result).toMatchSnapshot();
  });
});
