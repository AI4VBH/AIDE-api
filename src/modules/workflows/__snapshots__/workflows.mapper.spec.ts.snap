// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mapToPagedWorkflowsDto returns expected result 1`] = `
{
  "data": [],
  "totalPages": 1,
  "totalRecords": 0,
}
`;

exports[`mapToPagedWorkflowsItemDto returns expected result 1`] = `
{
  "ae_title": "",
  "data_origins": [
    "",
  ],
  "description": "Some description",
  "id": "fb94b9da-e957-4378-a626-8f0374e60cfb",
  "name": "Super Workflow",
  "revision": 2,
  "version": "1.0.0",
  "workflow_id": "c7610e0b-02b7-474d-b6e5-1224debc414d",
}
`;
